### ISSUE TITLE
[Rule]:  Input element with a list attribute and certain types shouldn't have a popup  

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#666]
See original opened [BUG] Should warn about aria-haspopup attribute for input role=combobox with list attribute #357

### Rule description

This rule is to make sure an <input> element with a list attribute (implied combobox) and certain input types shouldn't have a popup.   The requirement is here: https://w3c.github.io/html-aria/#el-input-text-list

### Relevant requirement(s)

502.3.1 Object Information

### High-level approach and rule scope

This rule is to check the input element that has type=text, search, tel, url, email, or with a missing or invalid type, and also a list attribute. The input element with the list attribute has an implicit role of combobox, and it shouldn't also use the aria-haspopup attribute.  
  id: "combobox_should_have_no_popup".  (?)
context: "input[list][aria-haspopup]"
 "aria_combobox_should_have_no_haspopup": {
        0: "An input element with a list attribute and with a type of text, search, tel, url, email, or a missing or invalid type shouldn't have a popup",
        "Pass": "An input element with a list attribute and with a proper type doesn't have a popup",
        "Potential_1": "The input element with a list attribute \"{0}\" and type \"{1}\" shouldn't also have an aria-haspopup attribute.
    }

ToolkitLevel:  LEVEL_ONE

Note that the other combobox rules (combobox_popup_reference, combobox_haspopup, combobox_focusable_elements, combobox_active_descendant, combobox_autocomplete) should depend on the current rule for short circuit. View details of the other combobox rule: https://github.com/IBMa/equal-access/blob/master/accessibility-checker-engine/src/v2/checker/accessibility/rules/rpt-combobox-rules.ts

### Ruleset and failure level

IBM, WCAG21, WCAG20

### Failure level
   
Needs review (Potential Violation)

### Failure cases
The first failure case is if the input element doesn't have the list attribute is not applicable.
The second case is an input element, with type=text, search, tel, url, email, or with a missing or invalid type, and a list attribute, if it also has an aria-haspopup attribute specified, then it is a potential violation that will need review by the developer to correct the intent.

**Comment by shunguoy** [comment]
initial dev branch: dev-357
